### PR TITLE
take standard gRPC ServerOption and expose gRPC Server

### PR DESCRIPTION
**Pull request by drewwells**
Enables passing grpc interceptors to themis so we can leverage the grpc ecosystem middleware like grpc_prometheus and grpc_logrus.

### CODE REVIEW

**Comment by daniel-garcia** [comment]
LGTM

File: pdpserver/server/server.go [hunk 1]
```diff
@@ -249,6 +258,11 @@ func NewServer(opts ...Option) *Server {
 	}
 }
 
+// GRPCServer expose the underlying gRPC Server
+func (s *Server) GRPCServer() *grpc.Server {
+	return s.requests.proto
```
Comment (drewwells) [inline pdpserver/server/server.go]: There's a race that causes this to be `nil` sometimes, we need to wait on a non-nil server

Comment (daniel-garcia) [reply pdpserver/server/server.go]: Looks like s.requests.proto is being recreated in LoadPolicies()

https://github.com/infobloxopen/themis/pull/245/files/32161ccd581a27f1984710426a54cf488aef7d92#diff-7e71dc72409d1564ddfa250aac5a9eabR556

Why can't the server be configured at Server creation time? What is changing during LoadPolicies that requires a NewGRPC server to be instantiated?

Comment (drewwells) [reply pdpserver/server/server.go]: it's not being re-created, it's just being created very late in `Serve()`. I'll see if I can get this moved to the appropriate place `NewServer()`

**Pull request merged**